api/client/run.go

```diff
@@ -228,6 +224,17 @@ func (cli *DockerCli) CmdRun(args ...string) error {
 
 	//start the container
 	if err := cli.client.ContainerStart(createResponse.ID); err != nil {
+		// If we have holdHijackedConnection, we should notify
+		// holdHijackedConnection we are going to exit and wait
+		// to avoid the terminal are not restored.
+		if config.AttachStdin || config.AttachStdout || config.AttachStderr {
+			select {
+			case <-exitCh:
+			default:
+				close(exitCh)
```

Comment:
@LK4D4  Just in case we may close channel somewhere in future and thus will leave a bug here, so it's much more safer to check if the channel has been closed before closing. but I'm ok with removeingthis check.